applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_custom_scipy_solver.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Import applications
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+
+# Import base class file
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_solver import MechanicalSolver
+
+from KratosMultiphysics import python_linear_solver_factory as linear_solver_factory
+
+# Import scipy modules
+import KratosMultiphysics.scipy_conversion_tools
+
+def CreateSolver(main_model_part, custom_settings):
+    return CustomScipySolver(main_model_part, custom_settings)
+
+class CustomScipySolver(MechanicalSolver):
+    """The structural mechanics custom scipy solver.
+
+    This class creates the mechanical solvers to provide mass and stiffness matrices as scipy matrices.
+    
+    Derived class must override the function SolveSolutionStep. In there the Mass and Stiffness matrices
+    can be obtained as scipy matrices.
+
+    See structural_mechanics_solver.py for more information.
+    """
+    def __init__(self, main_model_part, custom_settings):
+        # Construct the base solver.
+        super(CustomScipySolver, self).__init__(main_model_part, custom_settings)
+        KratosMultiphysics.Logger.PrintInfo("::[CustomScipySolver]:: ", "Construction finished")
+        self.space = KratosMultiphysics.UblasSparseSpace()
+
+    @classmethod
+    def GetDefaultSettings(cls):
+        this_defaults = KratosMultiphysics.Parameters("""{
+            "scheme_type"         : "dynamic"
+        }""")
+        this_defaults.AddMissingParameters(super(CustomScipySolver, cls).GetDefaultSettings())
+        return this_defaults
+
+    #### Private functions ####
+    def _create_solution_scheme(self):
+        """Create the scheme for the scipy solver.
+
+        The scheme determines the mass and stiffness matrices 
+        """
+        scheme_type = self.settings["scheme_type"].GetString()
+        if scheme_type == "dynamic":
+            solution_scheme = StructuralMechanicsApplication.EigensolverDynamicScheme()
+        else: # here e.g. a stability scheme could be added
+            err_msg =  "The requested scheme type \"" + scheme_type + "\" is not available!\n"
+            err_msg += "Available options are: \"dynamic\""
+            raise Exception(err_msg)
+
+        return solution_scheme
+
+    def _create_linear_solver(self):
+        ''' Linear solver will not be used. But eventually the solution stratgey calls the solver's clear function. 
+        To avoid crashing linear solver is provided here'''
+        return linear_solver_factory.CreateFastestAvailableDirectLinearSolver()
```

Comment:
if you only need a dummy then this should return the baseclass

Reply:
i agree